src/hash-table.js

```diff
@@ -7,6 +7,40 @@ class HashTable {
     this.storage = new LimitedArray(this.limit);
     // Do not modify anything inside of the constructor
   }
+  hash(data) {
+    let total = 0;
+    for (let i = 0; i < data.length; i++) total += data.charCodeAt(i);
+    return total % this.limit;
+  }
+  insert(key, value) {
+    const index = this.hash(key);
+    if (this.storage.storage[index] === undefined) {
```

Comment:
Start getting in the habit of using the `!` operator when you want to check if something is undefined or null. It is more concise and more flexible than checking if something is strictly just undefined.